distributed/worker.py

```diff
@@ -2209,9 +2214,17 @@ def memory_monitor(self):
                                 format_bytes(self.memory_limit))
                     break
                 k, v, weight = self.data.fast.evict()
+                del k, v
                 total += weight
                 count += 1
                 yield gen.moment
+                memory = proc.memory_info().rss
+                if total > need and memory > target:
+                    # Issue a GC to ensure that the evicted data is actually
+                    # freed from memory and taken into account by the monitor
```

Comment:
Is it expected for evicted data to have cyclic references?

Reply:
I don't think it's going to be the case for typical workloads but users are free to ship whatever data structures they like into `delayed` function calls. Why kind of problems do you anticipate?

Reply:
I simply think it's bad hygiene to add `gc.collect` calls "just in case" :-)

Reply:
For the record, this is the kind of actual case that was observed by dask-distributed users:

https://github.com/dask/zict/issues/19#issuecomment-313669395